spec/support/shared_examples_for_api.rb

```diff
@@ -1,44 +1,80 @@
-shared_examples "invalid API key validate endpoint" do
-  it "should return 401 and JSON error array when it is not provided" do
-    valid_params[:api_key] = nil
-    post_to_validate_endpoint
-    expect_unauthorised_error
-  end
+RSpec.shared_context 'deactivate deprecation warnings' do
+  before { allow(ActiveSupport::Deprecation).to receive(:warn) }
+end
 
-  it "should return 401 and JSON error array when it does not match an existing provider API key" do
-    valid_params[:api_key] = SecureRandom.uuid
-    post_to_validate_endpoint
-    expect_unauthorised_error
-  end
+RSpec.shared_examples "invalid API key validate endpoint" do |options|
+  include_context 'deactivate deprecation warnings'
 
-  it "should return 401 and JSON error array when it is malformed" do
-    valid_params[:api_key] = 'any-old-rubbish'
-    post_to_validate_endpoint
-    expect_unauthorised_error
+  context 'with invalid API key' do
+    it "response 401 and JSON error array when it is not provided" do
+      valid_params[:api_key] = nil
+      post_to_validate_endpoint
+      expect_unauthorised_error
+    end
+
+    it "response 401 and JSON error array when it does not match an existing provider API key" do
+      valid_params[:api_key] = SecureRandom.uuid
+      post_to_validate_endpoint
+      expect_unauthorised_error
+    end
+
+    it "response 401 and JSON error array when it is malformed" do
+      valid_params[:api_key] = 'any-old-rubbish'
+      post_to_validate_endpoint
+      expect_unauthorised_error
+    end
+
+    # TODO: it appears as though nested objects can be created on a claim by a
+    # provider other than that which created the claim
+    # - excluding for now
+    unless [:other_provider].include? options&.fetch(:exclude)
+      it "response 401 and JSON error array when it is an API key from another provider's admin" do
+        valid_params[:api_key] = other_provider.api_key
+        post_to_validate_endpoint
+        expect_unauthorised_error("Creator and advocate/litigator must belong to the provider")
+      end
+    end
   end
 end
 
-shared_examples "invalid API key create endpoint" do
-  it "should return 401 and JSON error array when it is not provided" do
-    valid_params[:api_key] = nil
-    post_to_create_endpoint
-    expect_unauthorised_error
-  end
+RSpec.shared_examples "invalid API key create endpoint" do |options|
```

Comment:
#not_a_blocker possible refactoring candidate?

Reply:
👍 Will create a ticket to include this refactoring, as discussed.